File: linkedlist/mergetwollsalternatively.js


function listNode(value){
    this.val = value;
    this.next = null
  }
  
  var p = new listNode(10)
  p.next = new listNode(20)
  p.next.next = new listNode(2)
  p.next.next.next = new listNode(6)

  var k = new listNode(1)
  k.next = new listNode(2)
  k.next.next = new listNode(4)
  k.next.next.next = new listNode(5)
  
  var linkedlist = function (a,b){
    var runner1 = a;
    var runner2 = b;
  
    if(!a){
      return b
    } 
    if(!b){
      return a
    }
    temp = a
    while(runner1 && runner2){
      a1 = runner1.next;
      b1 = runner2.next;
			runner1.next = runner2;
			runner2.next = a1;
			runner1 = a1;
			runner2 = b1;
    }
    return temp
    // console.log(runner2)
    
  }

  // var printLinkedList = function(p,k) {
  //     var t = p
  //     while(t) {
  //         console.log(t.val)
  //         console.log("---")
  //         t = t.next
  //     }
  // }
  
  console.log(linkedlist(p,k))

File: linkedlist/mergetwollsalternatively.py
class listNode:
  def __init__(self,val):
    self.val = val
    self.next = None

l1 =  listNode(10)
l1.next = listNode(20) 
l1.next.next = listNode(30) 

l2 =  listNode(10)
l2.next = listNode(20) 
l2.next.next = listNode(30) 

class LinkedList:
  def __init__(self):
    self.head = None

  def mergetwoLLs(self, head1, head2):
    if((not head1)):
      return head2
    
    if((not head2)):
      return head1

    runner = head1
    runner2 = head2
    while(runner and runner2):
      temp1 = runner.next
      temp2 = runner2.next
      runner.next = runner2
      runner2.next = temp1
      runner = temp1
      runner2 = temp2
    return head1

  def printList(self):
    temp = self.head
    while temp != None:
        print(temp.val)
        temp = temp.next

if __name__ == "__main__":
  print("welcome to python")
  list = LinkedList()
  list.mergetwoLLs(l1,l2)